securedrop/models.py

```diff
@@ -641,12 +641,19 @@ def login(cls,
               username: str,
               password: str,
               token: str) -> 'Journalist':
+
+        invalid_usernames = ['deleted']
+
         try:
             user = Journalist.query.filter_by(username=username).one()
         except NoResultFound:
             raise InvalidUsernameException(
                 "invalid username '{}'".format(username))
 
+        if user.username in invalid_usernames and user.uuid in invalid_usernames:
```

Comment:
We should raise if either username or UUID are invalid.

Reply:
@rmol I don't think this should have the `or`. What if a user already has an invalid username, `deleted` in this case. They should be able to login. What I understood from [this](https://github.com/freedomofpress/securedrop/issues/5232#issuecomment-624740370) was that we can differentiate legitimate users from `deleted` user using uuid, so we need to check both.

Reply:
Ah. Well, that's unfortunate, but yes, let's keep the `and`.